docs/heap_profiling.md

```diff
@@ -0,0 +1,140 @@
+# Heap Profiling
```

Comment:
Maybe rename to Device Memory Profiling, I think that'll be clearer to people browsing the docs.